x-pack/legacy/plugins/upgrade_assistant/public/np_ready/application/components/tabs/checkup/controls.tsx

```diff
@@ -32,44 +34,79 @@ export const CheckupControlsUI: FunctionComponent<CheckupControlsProps> = ({
+  const filterInvalid = Boolean(searchTermError);
+  return (
+    <EuiFlexGroup direction="column" responsive={false}>
+      <EuiFlexItem grow={true}>
+        <EuiFlexGroup alignItems="center" wrap={true} responsive={false}>
+          <EuiFlexItem grow={true}>
+            <EuiFieldSearch
+              isInvalid={filterInvalid}
+              aria-label="Filter"
+              placeholder={intl.formatMessage({
+                id: 'xpack.upgradeAssistant.checkupTab.controls.searchBarPlaceholder',
+                defaultMessage: 'Filter',
+              })}
+              onChange={e => {
+                const string = e.target.value;
+                const errorMessage = validateRegExpString(string);
+                if (errorMessage) {
+                  // Emit an empty search term to listeners if search term is invalid.
+                  onSearchChange('');
+                  setSearchTermError(errorMessage);
+                } else {
+                  onSearchChange(e.target.value);
+                  if (searchTermError) {
+                    setSearchTermError(null);
+                  }
+                }
+              }}
+            />
+          </EuiFlexItem>
 
-    {/* These two components provide their own EuiFlexItem wrappers */}
-    <FilterBar {...{ allDeprecations, currentFilter, onFilterChange }} />
-    <GroupByBar {...{ availableGroupByOptions, currentGroupBy, onGroupByChange }} />
+          {/* These two components provide their own EuiFlexItem wrappers */}
+          <FilterBar {...{ allDeprecations, currentFilter, onFilterChange }} />
+          <GroupByBar {...{ availableGroupByOptions, currentGroupBy, onGroupByChange }} />
 
-    <EuiFlexItem grow={false}>
-      <EuiButton
-        fill
-        onClick={loadData}
-        iconType="refresh"
-        isLoading={loadingState === LoadingState.Loading}
-      >
-        <FormattedMessage
-          id="xpack.upgradeAssistant.checkupTab.controls.refreshButtonLabel"
-          defaultMessage="Refresh"
-        />
-      </EuiButton>
-    </EuiFlexItem>
-  </EuiFlexGroup>
-);
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              fill
+              onClick={loadData}
+              iconType="refresh"
+              isLoading={loadingState === LoadingState.Loading}
+            >
+              <FormattedMessage
+                id="xpack.upgradeAssistant.checkupTab.controls.refreshButtonLabel"
+                defaultMessage="Refresh"
+              />
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+      {filterInvalid && (
+        <EuiFlexItem grow={false}>
+          <EuiCallOut
+            color="danger"
+            title={i18n.translate(
+              'xpack.upgradeAssistant.checkupTab.controls.filterErrorMessageLabel',
+              {
+                defaultMessage: 'Filter Invalid: {searchTermError}',
```

Comment:
nit: this should probably be sentence case per https://elastic.github.io/eui/#/guidelines/writing.